gt/lua_lpm.c

```diff
@@ -312,14 +323,14 @@ l_new_lpm6(lua_State *l)
 	lua_getfield(l, LUA_REGISTRYINDEX, GT_LUA_LCORE_ID_NAME);
 	lcore_id = lua_tonumber(l, -1);
 
-	p_lpm6 = lua_newuserdata(l, sizeof(struct rte_lpm6 *));
-	*p_lpm6 = init_ipv6_lpm("gt", &lpm6_conf,
+	lpm6_ud = lua_newuserdata(l, sizeof(struct lpm6_lua_userdata));
+	lpm6_ud->lpm6 = init_ipv6_lpm("gt", &lpm6_conf,
 		rte_lcore_to_socket_id(lcore_id), lcore_id,
 		rte_atomic32_add_return(&identifier6, 1));
-	if (unlikely(*p_lpm6 == NULL))
+	if (unlikely(lpm6_ud->lpm6 == NULL))
 		luaL_error(l, "gt: failed to initialize the IPv6 LPM table for Lua policies");
 
-	luaL_getmetatable(l, LUA_LPM6_TNAME);
```

Comment:
Bug: we are not saving `lpm6_conf` into `lpm6_ud`. This bug will go away once we eliminate the need for `lpm6_conf` as in the IPv4 case.